### ISSUE TITLE
Shelling out to collect environment variables is creating runaway 100% CPU processes on Linux

### ISSUE THREAD

**Issue by fooishbar** [atom/atom#13084]
### Prerequisites
- [x] Can you reproduce the problem in [safe mode](http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode)?
- [x] Are you running the [latest version of Atom](http://flight-manual.atom.io/hacking-atom/sections/debugging/#update-to-the-latest-version)?
- [x] Did you check the [debugging guide](http://flight-manual.atom.io/hacking-atom/sections/debugging/)?
- [x] Did you check the [FAQs on Discuss](https://discuss.atom.io/c/faq)?
- [x] Are you reporting to the [correct repository](https://github.com/atom/atom/blob/master/CONTRIBUTING.md#atom-and-packages)?
- [x] Did you [perform a cursory search](https://github.com/issues?q=is%3Aissue+user%3Aatom+-repo%3Aatom%2Felectron) to see if your bug or enhancement is already reported?
### Description

I'm fairly sure this will end up being an Electron issue, but here goes ...

When I start atom-beta (1.12.0b4, though was present with b3 at least), it spawns a shell ('env') subprocess which burns 100% CPU, attempting to read stdin forever:

```
systemd,1,1 --system --deserialize 126
  ├─atom,15392,10213
  │   ├─atom,15403,10213
  │   │   └─atom,15466,10213
  │   │       ├─zsh,15497,10213 -ilc command env
```

strace on the shell shows this looping forever:

```
rt_sigprocmask(SIG_SETMASK, [], [TSTP TTIN TTOU], 8) = 0
read(0, NULL, 0)                        = 0
rt_sigprocmask(SIG_BLOCK, [TSTP TTIN TTOU], [], 8) = 0
getpgrp()                               = 10213
ioctl(15, TIOCGPGRP, [11445])           = 0
getpgrp()                               = 10213
ioctl(15, TIOCGPGRP, [11445])           = 0
rt_sigprocmask(SIG_SETMASK, [], [TSTP TTIN TTOU], 8) = 0
read(0, NULL, 0)                        = 0
rt_sigprocmask(SIG_BLOCK, [TSTP TTIN TTOU], [], 8) = 0
```

Sometimes this process blocks Atom startup, and you have to kill the shell for Atom's window to show. Sometimes Atom works just fine despite this, and you only realise later when you wonder why your system is slow, and kill the extraneous process. Sometimes closing Atom takes the rogue shell process with it, and sometimes it outlives Atom. Sometimes everything works fine and I can't reproduce it at all.

Either way, safe mode or not has no effect on it, and nothing bright turns up with `--enable-logging`. Timecop shows 'window load time' taking an ungodly amount of time, but with no insight into why.

Happy to supply more details or attempt debugging as needed, with the caveat that I have no idea what I'm doing in JavaScript.
### Steps to Reproduce
1. Have my system
2. Run Atom beta

**Expected behavior:** 'What a nice normal text-editing experience this is'

**Actual behavior:** WOOOOOOOOSSHHHH goes the laptop fan
### Versions

Atom 1.12.0b4 (also occurred with b3)
Fedora 25 (kernel 4.8-rc8+, glibc 2.24, etc)

**Comment by thomasjo** [comment]
This originates in our [process environment patching](https://github.com/atom/atom/blob/d4d05244b313358c20717a7db07ec2085b5c39e3/src/update-process-env.js#L66). I'm guessing something is preventing `zsh` from exiting.

Would you mind running the follow in a terminal, and reporting back what the output is?

```
$ zsh -ilc command env
```

**Comment by fooishbar** [comment]
Oddly enough, just a screen-clear character, then exits immediately:
```
% zsh -ilc command env >foo 2>&1
% xxd foo
00000000: 1b5b 334a 1b5b 481b 5b32 4a              .[3J.[H.[2J
```

Thanks for the prompt response!

**Comment by fooishbar** [comment]
Ah, following the quoting for "command env" rather than as separate arguments, it dumps my entire environment and then exits immediately. I can share a redacted version with you, if the actual content is of interest, but it does simply print and exit.

**Comment by drewmnoel** [comment]
I have this problem as well. My atom won't even open until I `kill -9` the `zsh -ilc` process.

**Versions**:
```
$ atom --version
Atom    : 1.13.0-dev-630f146
Electron: 1.3.6
Chrome  : 52.0.2743.82
Node    : 6.3.0

$ apm --version
apm  1.14.0
npm  3.10.5
node 4.4.5
python 3.5.2
git 2.10.1
```

**Comment by TheRabbitologist** [comment]
I appear to have this issue as well.

When I launch atom-beta from a terminal instance, it runs without issues, regardless of whether it's launched in foreground mode or not, and whether I run the shell script in /usr/bin or the actual binary in /usr/share/atom-beta.

When I launch it through a provided desktop file, then the issue described in the bug report manifests. Running the exact same command used in the desktop file (minus the %F of course) does not reproduce this issue, and editing the desktop file to launch Atom using a simple `/usr/bin/atom-beta` does not fix it. Atom is the only application I have that causes this issue, including other Electron-based programs.

```
% atom-beta --version
Atom    : 1.12.0-beta3
Electron: 1.3.6
Chrome  : 52.0.2743.82
Node    : 6.3.0

% zsh --version
zsh 5.2 (x86_64-unknown-linux-gnu)

% cat /usr/share/applications/atom-beta.desktop
[Desktop Entry]
Name=Atom Beta
Comment=A hackable text editor for the 21st Century.
GenericName=Text Editor
Exec=env PYTHON=python2 XMODIFIERS= /usr/share/atom-beta/atom %F
Icon=atom-beta
Type=Application
StartupNotify=true
StartupWMClass=Atom
Categories=GNOME;GTK;Utility;TextEditor;Development;
MimeType=text/plain;
```

**Comment by 50Wliu** [comment]
@joefitzgerald

**Comment by bsandu** [comment]
Same issue here: "/bin/bash -ilc command env" keeps CPU in full load, and atom never starts. Only killing this will successfully load atom. 
Run on ubuntu 16.04
Atom version 1.12.0+1-1~webupd8~0

**Comment by dweldon** [comment]
I'm experiencing the same symptoms with atom `1.12.0` and bash `4.3.42` under linux mint.

**Comment by XzAeRo** [comment]
I'm experiencing the same.

I invoke `$ atom project/folder/`, but Atom only starts once I kill the `/bin/zsh -ilc command env` process. Otherwise Atom doesn't start and said process burns the CPU at 100% in all the cores.
Once the command is killed, Atom starts normally. For example:

```
$ atom project/folder/
# atom doesn't open
$ kill 12341
# burning process is killed and Atom opens its window
```

This is my installation, thorugh the webupd8 channel.
```
$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 16.04.1 LTS
Release:	16.04
Codename:	xenial

$ zsh --version
zsh --version
zsh 5.1.1 (x86_64-ubuntu-linux-gnu)

$ atom --version
Atom: 1.12.0
Electron: 1.3.6
Chrome: 52.0.2743.82
Node: 6.3.0

$ apm --version
apm  1.12.9
npm  3.10.5
node 4.4.5
python 2.7.12
git 2.7.4
```

**Comment by joefitzgerald** [comment]
For anyone reporting they are experiencing this issue, please run the relevant command:

* `/bin/bash -ilc "command env"`
* `/bin/bash -ilc "command env"`
* `<path-to-your-shell-here> -ilc "command env"`

You can find out what shell is being used by determining what the `SHELL` env variable is set to.

It sounds like we should at a minimum establish a timeout for the invocation of this command so that it will be killed if it takes too long.

**Comment by TheRabbitologist** [comment]
The "command env" commands (for both /bin/bash and /bin/zsh) both print a long list of environment variables without significant delay and exit normally. If there are any particular environment variables you wish to know the contents of, please let me know.

```
% which -a env
/usr/bin/env

% echo $SHELL
/bin/zsh

% getent passwd $LOGNAME | cut -d: -f7
/bin/zsh
```

**Comment by XzAeRo** [comment]
The command run on Atom invoke is `/bin/zsh -ilc command env` and behaves differently from `/bin/zsh -ilc "command env"`. The command Atom uses, outputs nothing, and the other one outputs all the environment variables. For example:

```bash
$ /bin/zsh -ilc "command env"
XDG_SEAT_PATH=/org/freedesktop/DisplayManager/Seat0
XDG_CONFIG_DIRS=/etc/xdg/xdg-ubuntu:/usr/share/upstart/xdg:/etc/xdg
SESSION=ubuntu
LANG=es_CL.UTF-8
DISPLAY=:0
SHLVL=1
LOGNAME=<user>
XDG_VTNR=7
MANDATORY_PATH=/usr/share/gconf/ubuntu.mandatory.path
GNOME_KEYRING_PID=
UPSTART_INSTANCE=
COMPIZ_CONFIG_PROFILE=ubuntu
XAUTHORITY=/home/<user>/.Xauthority
QT_LINUX_ACCESSIBILITY_ALWAYS_ON=1
QT_QPA_PLATFORMTHEME=appmenu-qt5
GTK_IM_MODULE=ibus
LANGUAGE=es_CL:es
XDG_SESSION_ID=c2
GTK2_MODULES=overlay-scrollbar
DESKTOP_SESSION=ubuntu
IM_CONFIG_PHASE=1
XDG_SESSION_DESKTOP=ubuntu
GDMSESSION=ubuntu
GNOME_KEYRING_CONTROL=
DEFAULTS_PATH=/usr/share/gconf/ubuntu.default.path
GNOME_DESKTOP_SESSION_ID=this-is-deprecated
UPSTART_EVENTS=xsession started
XDG_GREETER_DATA_DIR=/var/lib/lightdm-data/<user>
VTE_VERSION=4205
COMPIZ_BIN_PATH=/usr/bin/
PWD=/home/<user>
DBUS_SESSION_BUS_ADDRESS=unix:abstract=/tmp/dbus-P4IHCFTUWp
NODE_PATH=/usr/lib/nodejs:/usr/lib/node_modules:/usr/share/javascript
CLUTTER_IM_MODULE=xim
XDG_DATA_DIRS=/usr/share/ubuntu:/usr/share/gnome:/usr/local/share/:/usr/share/:/var/lib/snapd/desktop
SESSIONTYPE=gnome-session
QT4_IM_MODULE=xim
JOB=unity-settings-daemon
UPSTART_JOB=unity7
QT_ACCESSIBILITY=1
GDM_LANG=es_ES
QT_IM_MODULE=ibus
XDG_SESSION_TYPE=x11
SHELL=/bin/zsh
UPSTART_SESSION=unix:abstract=/com/ubuntu/upstart-session/1000/1719
TERM=xterm-256color
GTK_MODULES=gail:atk-bridge:unity-gtk-module
WINDOWID=56623114
XDG_CURRENT_DESKTOP=Unity
PATH=/home/<user>/bin:/home/<user>/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/home/<user>/.config/composer/vendor/bin:/home/<user>/.config/composer/vendor/bin
SSH_AUTH_SOCK=/run/user/1000/keyring/ssh
XDG_SEAT=seat0
HOME=/home/<user>
XMODIFIERS=@im=ibus
USER=<user>
XDG_SESSION_PATH=/org/freedesktop/DisplayManager/Session0
GPG_AGENT_INFO=/home/<user>/.gnupg/S.gpg-agent:0:1
XDG_RUNTIME_DIR=/run/user/1000
INSTANCE=
OLDPWD=/home/<user>
ZSH=/home/<user>/.oh-my-zsh
PAGER=less
LESS=-R
LSCOLORS=Gxfxcxdxbxegedabagacad
LS_COLORS=rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=00:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.Z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.jpg=01;35:*.jpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=00;36:*.au=00;36:*.flac=00;36:*.m4a=00;36:*.mid=00;36:*.midi=00;36:*.mka=00;36:*.mp3=00;36:*.mpc=00;36:*.ogg=00;36:*.ra=00;36:*.wav=00;36:*.oga=00;36:*.opus=00;36:*.spx=00;36:*.xspf=00;36:
_=/usr/bin/env
LC_CTYPE=es_CL.UTF-8

$ /bin/zsh -ilc command env
# empty output

$ which -a env
/usr/bin/env

$ echo $SHELL
/bin/zsh

$ getent passwd $LOGNAME | cut -d: -f7
/bin/zsh
```

Maybe it's something related to Atom expecting data?

**Comment by ak0ska** [comment]
Same as above. Atom 1.12.1 from the webupd8 PPA.
```
ahencz@ahencz-desktop:~$ atom --version
Atom    : 1.12.1
Electron: 1.3.6
Chrome  : 52.0.2743.82
Node    : 6.3.0

ahencz@ahencz-desktop:~$ which -a env
/usr/bin/env

ahencz@ahencz-desktop:~$ echo $SHELL
/bin/bash

ahencz@ahencz-desktop:~$ /bin/bash -ilc command env
# returns immediately, no output

ahencz@ahencz-desktop:~$ /bin/bash -ilc "command env"
# prints my environment variables with no delay
```

**Comment by fooishbar** [comment]
@drewmnoel Sorry, I missed your question earlier. Yes, starting with -f does seem to dodge the problem here, at least in a pretty limited sample size.

@XzAeRo I think Atom is doing the right thing; note that in `['-ilc', 'command env']` it passes `-ilc` as one argument and `command env` as the next argument (combined), which is exactly the same thing as `-ilc "command env"` from within a shell.

**Comment by benred42** [comment]
I appear to be having the same issue (same versions and output as @ak0ska) but running with -f does not get around the issue for me. Running `atom -f` returns:
```[30057:1111/090754:ERROR:browser_main_loop.cc(231)] Running without the SUID sandbox! See https://chromium.googlesource.com/chromium/src/+/master/docs/linux_suid_sandbox_development.md for more information on developing with the sandbox on.```

**Comment by gregpechiro** [comment]
I am having this [#13192](https://github.com/atom/atom/issues/13192) issue which could be the same (not sure). I am running Ubuntu 16.10

`atom -f` returns:

`[10254:1111/091653:ERROR:browser_main_loop.cc(231)] Running without the SUID sandbox! See https://chromium.googlesource.com/chromium/src/+/master/docs/linux_suid_sandbox_development.md for more information on developing with the sandbox on.`

If I start atom normally `atom .` It never runs. I opened my system monitor and found atom actually is running I'm just not seeing anything. If I close the terminal then atom starts fine. I can then open a new terminal and everything is working as it should.

Running `/bin/bash -ilc "command env"` returns:

`CLUTTER_IM_MODULE=xim
XDG_GREETER_DATA_DIR=/var/lib/lightdm-data/greg
GPG_AGENT_INFO=/home/greg/.gnupg/S.gpg-agent:0:1
ANDROID_HOME=/home/greg/android-sdk-linux
SHELL=/bin/bash
VTE_VERSION=4402
TERM=xterm-256color
SSH_AGENT_LAUNCHER=gnome-keyring
DERBY_HOME=/usr/lib/jvm/java-8-oracle/db
GRADLE_HOME=/home/greg/.sdkman/candidates/gradle/current
QT_LINUX_ACCESSIBILITY_ALWAYS_ON=1
WINDOWID=54525958
UPSTART_SESSION=unix:abstract=/com/ubuntu/upstart-session/1000/1650
SDKMAN_PLATFORM=Linux
GTK_MODULES=gail:atk-bridge
SDKMAN_CURRENT_API=https://api.sdkman.io/2
USER=greg
QT_ACCESSIBILITY=1
LS_COLORS=rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=00:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.Z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.jpg=01;35:*.jpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=00;36:*.au=00;36:*.flac=00;36:*.m4a=00;36:*.mid=00;36:*.midi=00;36:*.mka=00;36:*.mp3=00;36:*.mpc=00;36:*.ogg=00;36:*.ra=00;36:*.wav=00;36:*.oga=00;36:*.opus=00;36:*.spx=00;36:*.xspf=00;36:
SDKMAN_LEGACY_API=https://api.sdkman.io/1
XDG_SESSION_PATH=/org/freedesktop/DisplayManager/Session0
XDG_SEAT_PATH=/org/freedesktop/DisplayManager/Seat0
SSH_AUTH_SOCK=/run/user/1000/keyring/ssh
DEFAULTS_PATH=/usr/share/gconf/ubuntu.default.path
XDG_CONFIG_DIRS=/etc/xdg/xdg-ubuntu:/usr/share/upstart/systemd-session:/usr/share/upstart/xdg:/etc/xdg
PATH=/home/greg/bin:/home/greg/.local/bin:/home/greg/code/pebble-dev/arm-cs-tools/bin:/home/greg/code/pebble-dev/arm-cs-tools/bin:/home/greg/.sdkman/candidates/groovy/current/bin:/home/greg/.sdkman/candidates/gradle/current/bin:/home/greg/bin:/home/greg/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/usr/lib/jvm/java-8-oracle/bin:/usr/lib/jvm/java-8-oracle/db/bin:/usr/lib/jvm/java-8-oracle/jre/bin:/usr/local/go/bin:/home/greg/code/go/bin:/usr/lib/jvm/java-8-oracle:/home/greg/android-sdk-linux/tools:/home/greg/android-sdk-linux/platform-tools:/snap/bin:/usr/lib/jvm/java-8-oracle/bin:/usr/lib/jvm/java-8-oracle/db/bin:/usr/lib/jvm/java-8-oracle/jre/bin:/usr/local/go/bin:/home/greg/code/go/bin:/usr/lib/jvm/java-8-oracle:/home/greg/android-sdk-linux/tools:/home/greg/android-sdk-linux/platform-tools
DESKTOP_SESSION=ubuntu
QT_QPA_PLATFORMTHEME=appmenu-qt5
QT_IM_MODULE=ibus
PWD=/home/greg/code/go/src/github.com/cagnosolutions/cns
XDG_SESSION_TYPE=x11
JAVA_HOME=/usr/lib/jvm/java-8-oracle
XMODIFIERS=@im=ibus
LANG=en_US.UTF-8
MANDATORY_PATH=/usr/share/gconf/ubuntu.mandatory.path
SDKMAN_VERSION=5.1.4+78
GDM_LANG=en_US
IM_CONFIG_PHASE=1
COMPIZ_CONFIG_PROFILE=ubuntu
GDMSESSION=ubuntu
GTK2_MODULES=overlay-scrollbar
HOME=/home/greg
SHLVL=3
LANGUAGE=en_US
GOROOT=/usr/local/go
GNOME_DESKTOP_SESSION_ID=this-is-deprecated
LOGNAME=greg
XDG_SESSION_DESKTOP=ubuntu
GROOVY_HOME=/home/greg/.sdkman/candidates/groovy/current
SDKMAN_DIR=/home/greg/.sdkman
QT4_IM_MODULE=xim
XDG_DATA_DIRS=/usr/share/ubuntu:/usr/local/share/:/usr/share/:/var/lib/snapd/desktop
J2SDKDIR=/usr/lib/jvm/java-8-oracle
DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/1000/bus
GOPATH=/home/greg/code/go
MANAGERPID=1548
JOURNAL_STREAM=8:31774
SDKMAN_CANDIDATES_DIR=/home/greg/.sdkman/candidates
DISPLAY=:0
XDG_RUNTIME_DIR=/run/user/1000
XDG_CURRENT_DESKTOP=Unity
J2REDIR=/usr/lib/jvm/java-8-oracle/jre
GTK_IM_MODULE=ibus
COLORTERM=truecolor
XAUTHORITY=/home/greg/.Xauthority
_=/usr/bin/env`

**Comment by drewmnoel** [comment]
Interestingly, I tried the latest version but the shell process starts and exits well before the new timeout. Indeed, if I increase the timeout to 5 minutes it still starts normally. Rolling back to the version that was around when this issue was reported and **the problem reappears** (confirming not a node/zsh/whatever problem). I'll try completely reverting the commit to see if the problem somehow got solved elsewhere in the atom codebase. Very odd.

**Comment by pnicorelli** [comment]
```shell
willy@minerva:~$ atom -v
Atom    : 1.12.1
Electron: 1.3.6
Chrome  : 52.0.2743.82
Node    : 6.3.0
willy@minerva:~$ atom .
willy@minerva:~$ ps axcf | grep atom
 6026 pts/0    Rl     0:01  \_ atom
 6028 pts/0    R      0:00      \_ atom
 6060 pts/0    Sl     0:00      |   \_ atom
 6050 pts/0    Rl     0:00      \_ atom
```
shell vars:

```shell

willy@minerva:~$ /bin/bash -ilc "command env"
XDG_VTNR=2
LC_PAPER=it_IT.UTF-8
XDG_SESSION_ID=1
SSH_AGENT_PID=1495
LC_ADDRESS=it_IT.UTF-8
LC_MONETARY=it_IT.UTF-8
CLUTTER_IM_MODULE=xim
GPG_AGENT_INFO=/home/willy/.gnupg/S.gpg-agent:0:1
VTE_VERSION=4205
XDG_MENU_PREFIX=gnome-
SHELL=/bin/bash
TERM=xterm-256color
QT_LINUX_ACCESSIBILITY_ALWAYS_ON=1
GJS_DEBUG_OUTPUT=stderr
LC_NUMERIC=it_IT.UTF-8
WINDOWID=52428806
GJS_DEBUG_TOPICS=JS ERROR;JS LOG
GTK_MODULES=gail:atk-bridge
USER=willy
LC_TELEPHONE=it_IT.UTF-8
QT_ACCESSIBILITY=1
LS_COLORS=rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=00:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.Z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.jpg=01;35:*.jpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=00;36:*.au=00;36:*.flac=00;36:*.m4a=00;36:*.mid=00;36:*.midi=00;36:*.mka=00;36:*.mp3=00;36:*.mpc=00;36:*.ogg=00;36:*.ra=00;36:*.wav=00;36:*.oga=00;36:*.opus=00;36:*.spx=00;36:*.xspf=00;36:
SSH_AUTH_SOCK=/run/user/1000/keyring/ssh
DEFAULTS_PATH=/usr/share/gconf/gnome.default.path
USERNAME=willy
SESSION_MANAGER=local/minerva:@/tmp/.ICE-unix/1424,unix/minerva:/tmp/.ICE-unix/1424
XDG_CONFIG_DIRS=/etc/xdg/xdg-gnome:/etc/xdg
DESKTOP_SESSION=gnome
PATH=/usr/local/heroku/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin
QT_IM_MODULE=ibus
LC_IDENTIFICATION=it_IT.UTF-8
PWD=/home/willy
XDG_SESSION_TYPE=x11
XMODIFIERS=@im=ibus
LANG=en_US.UTF-8
MANDATORY_PATH=/usr/share/gconf/gnome.mandatory.path
LC_MEASUREMENT=it_IT.UTF-8
GDMSESSION=gnome
SHLVL=2
HOME=/home/willy
XDG_SEAT=seat0
GNOME_DESKTOP_SESSION_ID=this-is-deprecated
LOGNAME=willy
XDG_SESSION_DESKTOP=gnome
DBUS_SESSION_BUS_ADDRESS=unix:abstract=/tmp/dbus-RbHyxO2Ysn,guid=95ab28b54891454d8c95fc7658256daa
QT4_IM_MODULE=xim
XDG_DATA_DIRS=/usr/share/gnome:/usr/local/share/:/usr/share/:/var/lib/snapd/desktop
LESSOPEN=| /usr/bin/lesspipe %s
WINDOWPATH=2
DISPLAY=:1
XDG_RUNTIME_DIR=/run/user/1000
GTK_IM_MODULE=ibus
XDG_CURRENT_DESKTOP=GNOME
LC_TIME=it_IT.UTF-8
LESSCLOSE=/usr/bin/lesspipe %s %s
LC_NAME=it_IT.UTF-8
XAUTHORITY=/run/user/1000/gdm/Xauthority
_=/usr/bin/env
 ```

**Comment by joefitzgerald** [comment]
We are very interested in pairing / screen sharing with someone who is experiencing this issue with a hang on startup. There is a short term workaround to suppress the behavior introduced in https://github.com/atom/atom/pull/12562: set the `ATOM_DISABLE_SHELLING_OUT_FOR_ENVIRONMENT` environment variable to true.

**Comment by joefitzgerald** [comment]
Is there anyone experiencing this issue who is *not* running `linux`?

**Comment by as-cii** [comment]
It would also be helpful if someone could paste here the contents of `.bash_profile` (and `.bashrc`) or the equivalent for zsh. Basically, sharing any code that runs when shelling out to get the environment could help us track down the issue. Thanks! 🙇

**Comment by brychanrobot** [comment]
I'm getting the problem with a nearly vanilla mint .bashrc. I installed the image last week. So most things are pretty clean.

```
# ~/.bashrc: executed by bash(1) for non-login shells.
# see /usr/share/doc/bash/examples/startup-files (in the package bash-doc)
# for examples

# If not running interactively, don't do anything
case $- in
    *i*) ;;
      *) return;;
esac

# don't put duplicate lines or lines starting with space in the history.
# See bash(1) for more options
HISTCONTROL=ignoreboth

# append to the history file, don't overwrite it
shopt -s histappend

# for setting history length see HISTSIZE and HISTFILESIZE in bash(1)
HISTSIZE=1000
HISTFILESIZE=2000

# check the window size after each command and, if necessary,
# update the values of LINES and COLUMNS.
shopt -s checkwinsize

# If set, the pattern "**" used in a pathname expansion context will
# match all files and zero or more directories and subdirectories.
#shopt -s globstar

# make less more friendly for non-text input files, see lesspipe(1)
[ -x /usr/bin/lesspipe ] && eval "$(SHELL=/bin/sh lesspipe)"

# set variable identifying the chroot you work in (used in the prompt below)
if [ -z "${debian_chroot:-}" ] && [ -r /etc/debian_chroot ]; then
    debian_chroot=$(cat /etc/debian_chroot)
fi

# set a fancy prompt (non-color, unless we know we "want" color)
case "$TERM" in
    xterm|xterm-color|*-256color) color_prompt=yes;;
esac

# uncomment for a colored prompt, if the terminal has the capability; turned
# off by default to not distract the user: the focus in a terminal window
# should be on the output of commands, not on the prompt
#force_color_prompt=yes

if [ -n "$force_color_prompt" ]; then
    if [ -x /usr/bin/tput ] && tput setaf 1 >&/dev/null; then
	# We have color support; assume it's compliant with Ecma-48
	# (ISO/IEC-6429). (Lack of such support is extremely rare, and such
	# a case would tend to support setf rather than setaf.)
	color_prompt=yes
    else
	color_prompt=
    fi
fi

if [ "$color_prompt" = yes ]; then
    if [[ ${EUID} == 0 ]] ; then
        PS1='${debian_chroot:+($debian_chroot)}\[\033[01;31m\]\h\[\033[01;34m\] \W \$\[\033[00m\] '
    else
        PS1='${debian_chroot:+($debian_chroot)}\[\033[01;32m\]\u@\h\[\033[00m\] \[\033[01;34m\]\w \$\[\033[00m\] '
    fi
else
    PS1='${debian_chroot:+($debian_chroot)}\u@\h \w \$ '
fi
unset color_prompt force_color_prompt

# If this is an xterm set the title to user@host:dir
case "$TERM" in
xterm*|rxvt*)
    PS1="\[\e]0;${debian_chroot:+($debian_chroot)}\u@\h \w\a\]$PS1"
    ;;
*)
    ;;
esac

# enable color support of ls and also add handy aliases
if [ -x /usr/bin/dircolors ]; then
    test -r ~/.dircolors && eval "$(dircolors -b ~/.dircolors)" || eval "$(dircolors -b)"
    alias ls='ls --color=auto'
    #alias dir='dir --color=auto'
    #alias vdir='vdir --color=auto'

    alias grep='grep --color=auto'
    alias fgrep='fgrep --color=auto'
    alias egrep='egrep --color=auto'
fi

# colored GCC warnings and errors
#export GCC_COLORS='error=01;31:warning=01;35:note=01;36:caret=01;32:locus=01:quote=01'

# some more ls aliases
alias ll='ls -alF'
alias la='ls -A'
alias l='ls -CF'

# Add an "alert" alias for long running commands.  Use like so:
#   sleep 10; alert
alias alert='notify-send --urgency=low -i "$([ $? = 0 ] && echo terminal || echo error)" "$(history|tail -n1|sed -e '\''s/^\s*[0-9]\+\s*//;s/[;&|]\s*alert$//'\'')"'

# Alias definitions.
# You may want to put all your additions into a separate file like
# ~/.bash_aliases, instead of adding them here directly.
# See /usr/share/doc/bash-doc/examples in the bash-doc package.

if [ -f ~/.bash_aliases ]; then
    . ~/.bash_aliases
fi

# enable programmable completion features (you don't need to enable
# this, if it's already enabled in /etc/bash.bashrc and /etc/profile
# sources /etc/bash.bashrc).
if ! shopt -oq posix; then
  if [ -f /usr/share/bash-completion/bash_completion ]; then
    . /usr/share/bash-completion/bash_completion
  elif [ -f /etc/bash_completion ]; then
    . /etc/bash_completion
  fi
fi

if [ -x /usr/bin/mint-fortune ]; then
     /usr/bin/mint-fortune
fi

export GOPATH=$HOME/go
export PATH=$PATH:$GOPATH/bin
```

**Comment by nathansobo** [comment]
This should be fixed in 1.12.2 by making shelling out for the environment asynchronous and giving it a 5 second timeout.

**Comment by IronPeak** [comment]
Using
~~~~
$(atom .) &
~~~~
makes it startup without having to close the terminal.
Note: atom won't work on any of my machines (running 16.04 & 16.10).

**Comment by nathansobo** [comment]
@IronPeak So the changes in 1.12.2 didn't help at all?

**Comment by nathansobo** [comment]
Can you try `ATOM_DISABLE_SHELLING_OUT_FOR_ENVIRONMENT atom .`?

**Comment by IronPeak** [comment]
@nathansobo Sorry Nathan I forgot to mention I'm using ppa:webupd8team/atom,
just thought I would mention the work around I'm using while I wait for them to update.

Again sorry for the inconvenience.

**Comment by arcticicestudio** [comment]
Unfortunately I can not confirm that this bug is fixed in `1.12.2`.
I've compiled it from source using NodeJS version `6.9.1` (`7.x.x` fails early) and got the same 100% CPU burn.

**OS**: Arch Linux (GNOME `3.22`)
**Kernel**: `4.8.7-1`
**Atom**: `1.12.1`
**Electron**: `1.3.6`
**Chrome**: `52.0.2743.82`
**Node**: `6.3.0`

Got the same results in`--safe` mode. Furthermore there is no difference if it gets started via Launcher Icon (Plank or GNOME Dash Dock) or via terminal.

I was able to reproduce this with the following method:
  1. Start Atom via the launcher icon or from the terminal with `atom .` or `atom --new-window`
  2. Open the command palette and use *Window: Releaod" to reload the window
  3. Sometimes the window just stays completly white and unuseable, sometimes it reloads successfully
  4. Close the window and see your CPU go ham
  5. Try to open Atom via the launcher icon does nothing anymore

I have to kill my zsh process (`/bin/zsh -ilc command env`) to cool down my CPU again.
Also there is some curious behaviour if you follow these steps:
  1. Do step 1 - 5 described above
  2. After step 5 fire the `atom --new-window` command from your terminal (nothing will show up)
  3. Kill the CPU-eating process
  4. The moment the process gets killed Atom pops up (caused by the previously fired`atom --new-window` command)

**Comment by nathansobo** [comment]
@arcticicestudio Your version is `1.12.1`. Are you sure you compiled the right revision?

**Comment by arcticicestudio** [comment]
@nathansobo I'm sorry I was too tired and added the wrong patch number to the `PKGBUILD` file (`1` inseatd of `2` :sleepy: ).
I compiled version `1.12.2` for real now and can start Atom via the launcher icon without problems, **but** the 100% CPU bug still aperars if you reload the running Atom window. The process stays even if you close the Atom window.

**Comment by nathansobo** [comment]
Darn. I was really hoping the 5s timeout we added would kill the process if it went rogue. I'm going to rename this issue to describe the current state of affairs.

**Comment by drewmnoel** [comment]
If the shell process has not yet hit the 5 second timeout and you reload the window, it spawns a second shell process. The second one is killed after the 5 second timeout as expected. The first one is not killed and persists. By spamming the reload key shortcut, it is therefor possible to create more and more shell processes (just reload every 4 seconds). @arcticicestudio, is this the behavior you have as well?

**Comment by ronjouch** [comment]
I confirm the problem is only partially fixed in 1.13.0.beta2.

* Since beta2, Atom's main window appears, I see the child `zsh -ilc command env` process topping one of my cores and being killed after 5s. If I quit Atom before these 5s, the zsh process persists and I have to kill it manually.
* Everything @drewmnoel said about Window Reload behavior.
* Running `ATOM_DISABLE_SHELLING_OUT_FOR_ENVIRONMENT=true atom-beta` , no child zsh process is spawned, and nothing bad happens if I exit Atom before 5s.

Arch Linux, Linux 4.8.7, zsh 5.2, Atom 1.13.0.beta2 via the [AUR package](https://aur.archlinux.org/packages/atom-editor-beta/), Electron 1.3.6, Node 6.9.1

**Comment by joefitzgerald** [comment]
It seems like almost everyone still affected by this issue is running arch Linux. The changes we made allow Atom to start, but do not address the root cause: running "command env" should not hang indefinitely, and we need to observe this on a system that is actually affected.

If you were originally affected or continue to be affected, please go to http://atom-slack.herokuapp.com, sign into the atomio Slack team, and direct message @nathansobo or I with your details and your preferred mechanism for screen sharing.

Regarding the orphan process when a window is reloaded (this should be a fairly rare occupancy, for what it's worth), this can be fixed by calling `ChildProcess.kill` after subscribing to `process.on('kill', ...)`.

**Comment by arcticicestudio** [comment]
@drewchandler Yes, thats exactly the same behavior. The `ATOM_DISABLE_SHELLING_OUT_FOR_ENVIRONMENT=true atom` command seems to work for version `1.12.2` too.

**Comment by joefitzgerald** [comment]
@arcticicestudio are you an arch linux user also?

**Comment by arcticicestudio** [comment]
@joefitzgerald Yes, you can scroll up this site to see my system setup :smile: 
I would also like to help you with a screen share, but unfortunately my current location seem to be connected via bamboo internet lines which even makes google to load 10s+  :disappointed:

**Comment by drewmnoel** [comment]
There's a few Ubuntu users in here actually.

I'm at a complete loss. Converting the `execFile` call into the respective `exec` fixes the problem for me. That is, changing [update-process-env.js](https://github.com/atom/atom/blob/e34bc188d0c88c65fa9d3adfd8a4058b8e7f710d/src/update-process-env.js#L67):
```
-    childProcess.execFile(env.SHELL, ['-ilc', 'command env'], {encoding: 'utf8', timeout: 5000}, (error, stdout) => {
+    childProcess.exec('command env', {encoding: 'utf8', timeout: 5000}, (error, stdout) => {
```

If I change `env.SHELL` to be any typical command, it returns fine as an `execFile` call. If I change `env.SHELL` to be any shell (tried `bash`, `zsh`, `sh`), it hangs.

**Comment by drewmnoel** [comment]
Sanity question: why are we spawning the shell in interactive mode (i.e. `-i`)? Wouldn't that cause the shell to hang, waiting for input from stdin? This is **exactly** the behavior @fooishbar described in the original post.

**Comment by joefitzgerald** [comment]
We're trying to ensure the shell's environment is the result of running the same init scripts as the shell would run when a user launches an interactive shell.

**Comment by drewmnoel** [comment]
Isn't that why we're passing `-l` for a login shell? Passing `-i` informs it to read from stdin. In the `strace`, we see the shell repeatedly attempting to read from stdin. Removing this option dumps the same `env` that I expect.

**Comment by arcticicestudio** [comment]
Good point, next to this I can't see that there even is a `-l` flag for zsh.

The `-l` flag is bash-only to
> Make bash act as if it had been invoked as a login shell

**Comment by drewmnoel** [comment]
@arcticicestudio, actually `zsh` has support for `-l` as well since at least 2002.

**Comment by nathansobo** [comment]
Can someone hack the different flag into Atom and see if it addresses the issue?

**Comment by joefitzgerald** [comment]
The changes have been applied in https://github.com/atom/atom/pull/13213. Try testing from that branch...

**Comment by drewmnoel** [comment]
That should do it. Works the same as my commit did, and also should take care of any stragglers in the event something goes wrong.

**Comment by joefitzgerald** [comment]
Hopefully this article describes why we use the interactive flag: https://coderwall.com/p/u003pa/bash-startup-scripts-on-linux-and-mac-os-x

On OS X and on Linux, this causes `.bashrc` (and similar) scripts to be run by the shell.

I'm wondering if your `.zshrc` is including your `.zshprofile` and doing vice versa? This would cause an infinite loop of inclusion, which would replicate this behavior?

**Comment by drewmnoel** [comment]
Interactive shells are shells that assume interaction. Login shells are shells that source the relevant files. From the link you posted, we should just require that the shell is a login shell (`-l`):

> Here's the standard behavior: login shells always look for the bash configuration files with "profile" in the name, in this order: /etc/profile, ~/.bash_profile, then ~/.bash_login and lastly ~/.profile. When login shells exit, they read ~/.bash_logout.

I don't have any sourcing loops, or running the command manually would cause the same issue.

**Comment by joefitzgerald** [comment]
This behavior was introduced for two reasons:

1. On OS X, launching atom from Dock / Finder / Spotlight results in an anemic environment for which no shell initialization scripts have been run
1. On Linux, launching atom from Unity Launcher (or similar) results in the environment that one would expect from a login shell, but *not* the environment one would expect from an interactive login shell

Let's put OS X aside now, because it is orthogonal (but we must retain the `-i` flag for it).

On Linux, many users don't _fully_ understand the way lifecycle for their shell initialization scripts. They launch atom from Unity Launcher on Ubuntu and wonder "why is my environment different to the environment in my terminal??". And then they blame Atom. Those who understand the full lifecycle immediately jump to the `rc` / `profile` scripts to fix the dichotomy.

Our goal is to ensure both `rc` and `profile` scripts are run for users on Linux and OS X, regardless of where they launch atom. We have modified https://github.com/atom/atom/blob/master/atom.sh#L18 to disable shelling out for the environment because implicitly, launching atom from `atom.sh` means you're in an interactive [login] shell.

I have observed on arch linux via the non-official distribution of Atom, this script is not correct and the env variable is not being set. This is the source of part of the issue. The other part of the issue is that in some circumstances using the `-i` flag _does_ work on linux. It most certainly works on OS X and has been for a long time. It's not clear to me _why_ it is not working on linux _some_ of the time. I can only surmise that it is occurring when Atom has already been launched from a terminal.

**Comment by lindem** [comment]
As an immediate remedy which is not fixing the problem, I have resorted to starting atom via a small shell script killing all such bash processes after three seconds, which is a reliable workaround and will likely do for a while. 
```
#! /bin/bash

ATOM_DISABLE_SHELLING_OUT_FOR_ENVIRONMENT=true /usr/bin/atom "$@" &

## wait a few secs, then kill all env collecting processes
sleep 3
PIDS=`ps aux | grep 'bash -ilc command env' | awk '{print $2}'`

for i in $PIDS
do
	kill -9 $i
	echo "killed: $i"
done

echo "killed all processes"
```

**Comment by as-cii** [comment]
We will release a hotfix on stable and beta later today which should address this. Thanks everyone for the feedback!

**Comment by maxbrunsfeld** [comment]
Hey everyone; I think this is fixed in v1.12.3 and v1.13.0-beta3, thanks to @joefitzgerald.

**Comment by joefitzgerald** [comment]
I would caveat that there is an outstanding issue with the arch Linux unofficial AUR package; this uses an atom.sh that is missing the export of a key environment variable.

**Comment by ronjouch** [comment]
> I would caveat that there is an outstanding issue with the arch Linux unofficial AUR package; this uses an atom.sh that is missing the export of a key environment variable.

Thanks for the work, everyone. I just notified [the AUR package](https://aur.archlinux.org/packages/atom-editor-beta/) maintainer about beta3 and asked to add this export and why we don't use upstream's script rather than a forked version.

**Comment by arcticicestudio** [comment]
Good job everyone. Maybe a official `PKGBUILD` would prevent feature problems like this?
It could be a template which gets populated with the correct values by the build script. This could also be added to the list of attachments for each release so the AUR maintainer can simply grab it.
I'm used to create my own `PKGBUILD` files anyway, but as a Arch Linux :heart:  I'd like to enhance the update process for Atom.

I can create a new issue if you think this idea should be added to the backlog :smile:

**Comment by nathansobo** [comment]
It would have to come via a pull request that also explains what a `PKGBUILD` is and ensures it doesn't rot by making it part of the build in some way.

**Comment by ronjouch** [comment]
> Maybe a official PKGBUILD would prevent future problems like this?

Not sure this is a good idea:

1. That would introduce friction to the packager's job.
2. That would deviate from regular in-archlinux.org Git versioning of PKGBUILDs.

The initial problem was that `atom-editor-beta` did fork/replicate the `atom.sh` script. And after mention to the current maintainer [he removed it minutes ago](https://aur.archlinux.org/cgit/aur.git/commit/?h=atom-editor-beta&id=898040b0677e3efdf421fd97716c362a6b0a9f2c), reverting to the upstream version :) . I haven't tested the new PKGBUILD yet, but it looks all good to me.

**Comment by gregpechiro** [comment]
Ok I know this issue has been marked closed but I still seem to have this problem AFTER upgrading to 1.12.3. If I launch atom from the terminal `atom .` the atom window opens but no files are loaded. I then run `htop` in a separate terminal and 2 of my cores are pegged at 100%. If I close the terminal then the cores go down to normal atom loads the files. Thoughts? Is it just me?

**Comment by DisasteR** [comment]
I can confirm that problem is still here.

Running Ubuntu 16.04.1 LTS

Atom    : 1.12.3
Electron: 1.3.6
Chrome  : 52.0.2743.82
Node    : 6.3.0

from ppa webupd8team/atom

**Comment by joefitzgerald** [comment]
@gregpechiro are you also using an unofficial distribution of atom (e.g. ppa webupd8team/atom)?

**Comment by gregpechiro** [comment]
Yes I am

**Comment by DisasteR** [comment]
as described by @gregpechiro 

opened using `atom .`
and resolve when i close terminal used to open atom

**Comment by bsandu** [comment]
+1 to @gregpechiro. Also, using webupd8team/atom. Installed version is 1.12.3-1~webupd8~0.

**Comment by thomasjo** [comment]
The webupd8 distribution of Atom is broken. They are shipping an outdated version of `atom.sh`...

**Comment by 50Wliu** [comment]
Please uninstall Atom if you're using the webupd8 PPA and reinstall it from https://atom.io.

For more information see https://github.com/atom/design-decisions/blob/master/unofficial-atom-distributions.md.

**Comment by gregpechiro** [comment]
Thanks for the help. I uninstalled atom and removed the repo. Then downloaded the .deb from atom,io and installed. It is now working fine. Sorry I didn't realize webupd8team/atom was not the mainstream version. I guess I didn't look into it well enough and assumed it was just a repository for the atom master branch maintained by a third party.

**Comment by XzAeRo** [comment]
I just updated to the latest webupd8 version (`atom 1.12.3-1~webupd8~0`) , and it's working properly by the way.

**Comment by DisasteR** [comment]
i have contacted webupd8 to report that issue.
They have just  updated the package with latest `atom.sh`

**Comment by dw-thomast** [comment]
Hi everyone, I still have the same problem.

Running Ubuntu 16.04.1 LTS

Atom : 1.12.3  (just uninstall and install it again)
Chrome : 54.0.2840.100
Node : 5.11.1

My CPU load due to this command
> xxxx  10445 99.6  0.0  17556  1236 tty2     R    14:57   0:27 /usr/bin/zsh -ilc command env

This command works well
> /bin/zsh  -ilc "command env"

I need to start my atom with this command : 
> OM_DISABLE_SHELLING_OUT_FOR_ENVIRONMENT=true atom

**Comment by joefitzgerald** [comment]
@dw-thomast are you using the webupd8team PPA? Also, what version of zsh are you using?

**Comment by dw-thomast** [comment]
Before I read this thread, I was using the PPA version. Now I'm using the official version. I purge my previous version before install the new one. But I keep mon config files

Terminator: 0.98-1
Zsh: 5.1.1-1ubuntu2
Oh-my-version: up to date (3477ff25274fa75bd9e6110f391f6ad98ca2af72)

**Comment by joefitzgerald** [comment]
Could you update zsh to v5.2 or greater?

**Comment by joefitzgerald** [comment]
https://github.com/zsh-users/zsh-autosuggestions/issues/24

Also make sure you are not using zsh-autosuggestions.

**Comment by dw-thomast** [comment]
Thks for your advices.
I will try to update my zsh asap

> apt-cache policy zsh
> zsh:
>   Installé : 5.1.1-1ubuntu2
>   Candidat : 5.1.1-1ubuntu2
>  Table de version :
>  *** 5.1.1-1ubuntu2 500
>         500 http://fr.archive.ubuntu.com/ubuntu xenial/main amd64 Packages
>         100 /var/lib/dpkg/status

**Comment by forivall** [comment]
Note that I'm still having this issue:

Arch Linux
zsh 5.2
atom 1.12.4
It occurs with either the package from arch's official community repo (https://git.archlinux.org/svntogit/community.git/tree/trunk?h=packages/atom) or from the `atom-editor-bin` package (https://aur.archlinux.org/packages/atom-editor-bin/ / https://aur.archlinux.org/cgit/aur.git/tree/?h=atom-editor-bin).

Note that it only happens when I launch atom from the menu in gnome. If I launch it from a terminal line, it works perfectly fine (with the aur `-bin` package, i haven't tested with the community package).

Look at the .desktop file for how atom is launched from the menu. Should the menu launcher also launch from the `atom.sh` script?

**Comment by forivall** [comment]
Minimal reproduction case: https://gist.github.com/forivall/99bd9b913ab774ced025be2483a59879
Usage: `xdg-launch.py ./spawn-ilc.desktop`. You'll need to modify the path in the .desktop file

Now to figure out if this is a gnome or node issue. Changing it from zsh to bash causes the same issue.

**Comment by joefitzgerald** [comment]
Just to confirm, @forivall, you're not using an official distribution of atom from https://atom.io?

https://github.com/atom/design-decisions/blob/master/unofficial-atom-distributions.md

**Comment by forivall** [comment]
@joefitzgerald the aur package `atom-editor-bin` downloads the official distribution, makes minimal tweaks for arch, and then installs it globally, just as if it were installed on ubuntu or fedora.

If someone else has the time to try out my reproduction case on ubuntu, that would be great.

the following gist makes it work, by using a pty for stdin:
https://gist.github.com/forivall/825f4a94eaf30b96dec522d40bb47887

**Comment by joefitzgerald** [comment]
@forivall I appreciate your effort in trying to identify root cause. It does seem like something very specific to arch linux at this point, and it seems like you are on the right track to look at the method of invocation for non-terminal launched atom on arch.

Very interested to hear the outcome of your investigation!

**Comment by forivall** [comment]
I started up an ubuntu 16.10 vm: with the minimal reproduction, the same issue occurs, and the same fix works. :tada:

**Comment by GabMus** [comment]
Hello, I'm having the same issue with the zsh process taking 100% of the CPU. I tested it in Arch Linux, using the official community repo package, downloading the tar.gz from atom release page for the stable and the beta versions.

All of these versions share the same problem, and weirdly enough it only happens when launching atom from a .desktop file (via gnome shell in my case) and doesn't when I launch atom from the command line.

After reading @arcticicestudio comment in #13279 I came up with a sort of workaround, I simply made a .desktop file with the following line: `Exec=sh -c "ATOM_DISABLE_SHELLING_OUT_FOR_ENVIRONMENT=true atom"` and it seems to work fine.

I hope this can be of help to find a long term solution to this problem, which is getting really annoying and prevents many people from using such a great editor as atom.

**Comment by nathansobo** [comment]
@joefitzgerald should we reopen this issue? Seems like we're still getting reports but it's really hard to tell what's real with all these different custom packaging solutions.

**Comment by thomasjo** [comment]
@nathansobo We should at least attempt to reproduce the reported problems using only _official_ distributions of Atom.

This means we'll need to check Arch Linux at least, possibly with both GNOME and KDE. On the Ubuntu side of things, it might be enough to check Ubuntu GNOME 16.04. The only reported problems I've seen on Ubuntu 16.04 is in relation to webupd8.

**Comment by phw** [comment]
I debugged this a bit and can reproduce it with the following test script (extracted from [src/update-process-env.js](https://github.com/atom/atom/blob/1b2d2f029945c58cf4cd474ef1423856b35cf26c/src/update-process-env.js)):

```javascript
const childProcess = require('child_process')
const cmd = 'command env'
const shell = process.env.SHELL  // zsh or bash doesn't matter, behavior is the same
// const shell = '/bin/bash'

const child = childProcess.spawn(shell, ['-ilc', cmd], {
   encoding: 'utf8',
  //  detached: true,
  stdio: ['ignore', 'pipe', process.stderr]
})
child.on('error', (e) => {
  console.log('error', e)
})
child.stdout.on('data', (data) => {
  console.log(data.toString('utf8'))
})
child.on('close', (code, signal) => {
  console.log('close', code, signal)
})
```

Launching this directly from a shell works, even if I set an empty environment like so:

    env -i SHELL=/bin/bash node spawn.js

Launching the same from a desktop file in Gnome 3.22 leads to the shell task endlessly consuming 100% of CPU. I used the following desktop file:

```
[Desktop Entry]
Name=Spawn Node Test
Exec=env SHELL=/bin/bash node /home/phw/devel/spawn.js
Icon=atom
Type=Application
```

Two things help to solve this issue:

a) Enable `detached: true` when spawning the child process
b) Remove the `-i` flag from the shell call

I think b) would actually be correct anyway, since this is in no way an interactive shell.

**Comment by joefitzgerald** [comment]
@maxbrunsfeld guess we want to add detached back in.

@phw thank you for the detailed analysis. We do actually want the -i flag as the intent of this behavior is to ensure the shell initializatoon scripts that get run for an interactive shell are run. This ensures that atom has the same environment that it would have when invoked from a terminal.

**Comment by phw** [comment]
Can this issue be reopened then until a fix is merged? I think this was accidentally closed by https://github.com/atom/atom/pull/13283

**Comment by joefitzgerald** [comment]
No, it wasn't closed by that. We should open a new issue with your findings @phw. I will commit to getting this in today.
